conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/Endpoints.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.conjure.java.undertow.lib.Endpoint;
+import io.undertow.server.HandlerWrapper;
+
+public final class Endpoints {
+
+    private Endpoints() {}
+
+    /**
+     * Build am {@link Endpoint} who is a copy of the {@link Endpoint} but whose handler
+     * has been wrapped by an {@link HandlerWrapper}.
+     */
+    public static Endpoint mapHandler(Endpoint endpoint, HandlerWrapper wrapper) {
```

Comment:
Thoughts on naming this method `wrap`? It doesn't exclude us from eventually adding a `wrap` method for the full endpoint because the parameter types will differ.

Reply:
wrap seems to me to only makes sense when you are returning the full object wrapped itself.
The `map` here comes from the fact that Endpoint is a `Functor` (actually a x-functor where x is the number of fields). I like `map` here :)

Reply:
In that case can we rename the method from `mapHandler` to `map` since the paramter types tell us that it's only capable of mapping handlers?

Reply:
Sure.

Reply:
I thought it was less confusing with `mapHandler` but renamed to `map`